extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -408,4 +413,179 @@ private static Table readDataIndexTable(
+                if (rs.isEmpty()) {
+                    continue;
+                }
+
+                consumer.accept(rgIdx, rs);
+            }
+        }
+    }
+
+    @Override
+    public PushdownResult pushdownFilter(
+            final Map<String, ColumnSource<?>> columnSourceMap,
+            final WhereFilter filter,
+            final RowSet input,
+            final boolean usePrev,
+            final FilterContext context,
+            final long costCeiling) {
+
+        final long executedFilterCost = context.executedFilterCost();
+
+        final List<String> parquetNames = filter.getColumns().stream().map(
+                readInstructions::getParquetColumnNameFromColumnNameOrDefault).collect(Collectors.toList());
+        final List<Integer> parquetIndices = parquetNames.stream().map(
+                name -> parquetMetadata.getFileMetaData().getSchema().getFieldIndex(name)).collect(Collectors.toList());
+
+        final RowSetBuilderSequential matchBuilder = RowSetFactory.builderSequential();
+        final RowSetBuilderSequential maybeBuilder = RowSetFactory.builderSequential();
+
+        // Should we look at the metadata?
+        if (!QueryTable.DISABLE_WHERE_PUSHDOWN_METADATA && executedFilterCost < PushdownResult.METADATA_STATS_COST) {
+            // Some range filter host a condition filter as the internal filter and we can't push that down.
+            if (filter instanceof RangeFilter
+                    && ((RangeFilter) filter).getRealFilter() instanceof AbstractRangeFilter) {
+                final AbstractRangeFilter rf = (AbstractRangeFilter) ((RangeFilter) filter).getRealFilter();
+                // Only one column in a RangeFilter
+                final Integer parquetIndex = parquetIndices.get(0);
+
+                iterateRowGroupsAndRowSet(input, (rgIdx, rs) -> {
+                    final Pair<Object, Object> p = getMinMax(
+                            parquetMetadata.getBlocks().get(rgIdx).getColumns().get(parquetIndex).getStatistics());
+
+                    if (p == null || rf.overlaps(p.first, p.second)) {
+                        maybeBuilder.appendRowSequence(rs);
+                    }
+                });
+                return PushdownResult.of(matchBuilder.build(), maybeBuilder.build());
+            }
+            if (filter instanceof MatchFilter) {
+                final MatchFilter mf = (MatchFilter) filter;
+
+                // Only one column in a MatchFilter
+                final Integer parquetIndex = parquetIndices.get(0);
+
+                iterateRowGroupsAndRowSet(input, (rgIdx, rs) -> {
+                    final Pair<Object, Object> p = getMinMax(
+                            parquetMetadata.getBlocks().get(rgIdx).getColumns().get(parquetIndex).getStatistics());
+
+                    if (p == null) {
+                        maybeBuilder.appendRowSequence(rs);
+                        return;
```

Comment:
Are statistics universal?  If not maybe we should have a case where the statistics are not present on one partition to cover this case and the == null on 552.